test/tracer/tracer.c

```diff
@@ -77,6 +89,49 @@ void* thread2() {
   return 0;
 }
 
+void check_lw_sign_ext() {
+#ifdef RV64G
+  volatile uint64_t  mem[2]   = { 0 };
+  volatile uint64_t* p        = &mem[0];
+  volatile uint64_t  badneg   = 0xbadbeef1badbeef0;
+  volatile uint64_t  checkneg = 0xffffffffbadbeef0;
+  TRACE_PUSH_ON
+lw_neq:
+  asm volatile( "ld   t3, 0(%1)   \n\t"     // t3 <- badneg
+                "ld   t4, 0(%2)   \n\t"     // t4 <- checkneg
+                "sd   t3, 4(%0)   \n\t"     // store badneg to upper half of mem[0]
+                "lw   t3, 0(%0)   \n\t"     // load mem[0]
+                "beq  t3, t4, _okneg \n\t"  // check
+                ".word 0x0 \n\t"            // fail
+                "_okneg: \n\t"
+                "sd   t3, 0(%0)  \n\t"  // store back into mem
+                : "=r"( p )
+                : "r"( &badneg ), "r"( &checkneg )
+                : "t3" );
+  printf( "neg result is 0x%llx\n", *p );
```

Comment:
`printf("neg result is 0x%" PRIx64 "\n", *p);` should be used instead.

`#include <inttypes.h>` to get the `PRI` and `SCN` type-specific format macros.